### PR TITLE
Allow customizing order of bify transforms/plugins

### PR DESCRIPTION
**Pull request by mcmire**
We use Browserify to compile our JavaScript code into a distributable
format. You can use Browserify in a few different ways, but we use the
API, which is detailed [here][1]. The way we use it is as follows:

``` javascript
const bundlerOpts = {
  entries: [],
  transform: [],
  plugin: [],
  require: [],
  manualExternal: [],
  manualIgnore: [],
}
// ... change bundlerOpts in various ways ...

// then, later:
const bundler = browserify(bundlerOpts);
bundler.bundle();
```

Note that all transforms and all plugins are passed to `browserify` in
one go. This creates a problem, because transforms and plugins are
processed in the same order in which they are specified. That is,
specifying transform A, then plugin B, then transform C, is NOT the same
as specifying transform A, then transform C, then plugin B.

The way to correct this is to make use of Browserify's full API, which
provides `transform` and `plugin` methods, as well as other specific
methods to specify other options. This commit, then, updates the part of
the build system that configures Browserify to use this API.

The order dependency as it relates to transforms and plugins will come
into play when introducing TypeScript into the build system. Namely, we
will be introducing a `tsify` Browserify plugin, which will need to be
specified before specifying `babelify`, as demonstrated [here][2]. Hence
we need this commit to be in place so that this is possible.

[1]: https://www.npmjs.com/package/browserify#browserifyfiles--opts
[2]: https://github.com/TypeStrong/tsify#es2015-formerly-known-as-es6

---

Manual testing steps:  

* Make sure you've added the development version of the extension to your browser (sourced from `dist/`).
* Run `yarn start`. Reload the extension if need be. You should not see a white screen. Click around in the extension for a bit. You should not encounter any error screens.
* Run `yarn start:test`. Run an e2e test. It should pass.
* Run `yarn dist`. Reload the extension if need be. You should not see a white screen. Click around in the extension for a bit. You should not encounter any error screens.

### CODE REVIEW

File: development/build/scripts.js
Comment (mcmire) [inline development/build/scripts.js:690]: This is where `tsify` will go. According to the [documentation](https://github.com/TypeStrong/tsify#es2015-formerly-known-as-es6), it needs to go before `babelify`.

File: development/build/scripts.js
Comment (mcmire) [inline development/build/scripts.js:808]: I didn't see a real reason this was nested within `bundleIt`, and considering most of `bundleIt` is this function anyway, I pulled it out.

**Review by digiwand — DISMISSED** [review]
Lovely solution! More modular, succint, and got rid of the "manual" code 👍

File: development/build/scripts.js
Comment (mcmire) [reply development/build/scripts.js:355]: I went back and forth on that. If it's singular and you pass multiple arguments, then it looks like `addPlugin(plugin1, plugin2, plugin3)`. If it's plural and you pass one argument, then it looks like `addPlugins(plugin1)`. Do you think the second option looks better than the first?

Comment (digiwand) [reply development/build/scripts.js:355]: Oh, I see where you're coming from. Also, when we've used the pluralized version in other places in this file, we used it to pass in arrays.

I like the singular version now that you wrote it out like that, and I've reconsidered! Thanks for taking the time to think this out with me! I like the existing decision you put in place here.

`addPlugin(plugin1, plugin2, plugin3)` `addPlugin(plugin1)` ✅ 
`addPlugins(plugin1, plugin2, plugin3)` `addPlugins(plugin1)`❌

File: development/build/scripts.js
Comment (digiwand) [reply development/build/scripts.js]: No need for any changes! Thanks for supporting this thought! haha. Yes, I also agree TypeScript can make this relatively easier to read. Looking forward to adding more TypeScript into the codebase!

**Comment by mcmire** [comment]
@digiwand Made a small change in response to your feedback. Can you take another quick look?

File: development/build/scripts.js
Comment (digiwand) [reply development/build/scripts.js]: *sees [b098f74](https://github.com/MetaMask/metamask-extension/pull/13483/commits/b098f748242f85105ebe471bd599dd28b2a3efed) just as I post the last comment*
*thinks, "what have I done"* 🤣

 Thanks! I appreciate it! haha LGTM!

**Review by digiwand — APPROVED** [review]

**Comment by mcmire** [comment]
This PR may no longer be necessary since we're no longer adding `tsify`. I'll wait until #13495 is merged.

**Comment by mcmire** [comment]
We won't need this after all so I am closing this.

**Pull request closed without merging**
